sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1288,4 +1299,71 @@ export class ApplicationService extends RecordDataModelService<Application> {
     }
     return applicationQuery.getOne();
   }
+
+  /**
+   * Validation for application overlapping dates or Pending PIR.
+   * This validation can be disabled by setting BYPASS_APPLICATION_SUBMIT_VALIDATIONS to true in .env file.
+   * @param applicationId
+   * @param lastName
+   * @param userId
+   * @param sin
+   * @param birthDate
+   * @param studyStartDate
+   * @param studyEndDate
+   * @param bypassApplicationSubmitValidations
+   */
+  async validateOverlappingDatesAndPIR(
+    applicationId: number,
+    lastName: string,
+    userId: number,
+    sin: string,
+    birthDate: Date,
+    studyStartDate: Date,
+    studyEndDate: Date,
+  ): Promise<void> {
+    if (!this.bypassApplicationSubmitValidations) {
+      const existingOverlapApplication = this.validatePIRAndDateOverlap(
+        userId,
+        new Date(studyStartDate),
+        new Date(studyEndDate),
+        applicationId,
+      );
+
+      const existingSFASFTApplication =
+        this.sfasApplicationService.validateDateOverlap(
+          sin,
+          birthDate,
+          lastName,
+          new Date(studyStartDate),
+          new Date(studyEndDate),
+        );
+
+      const existingSFASPTApplication =
+        this.sfasPartTimeApplicationsService.validateDateOverlap(
+          sin,
+          birthDate,
+          lastName,
+          new Date(studyStartDate),
+          new Date(studyEndDate),
+        );
+      const [
+        applicationResponse,
+        sfasFTApplicationResponse,
+        sfasPTApplicationResponse,
+      ] = await Promise.all([
+        existingOverlapApplication,
+        existingSFASFTApplication,
+        existingSFASPTApplication,
+      ]);
+      if (
+        !!applicationResponse ||
+        !!sfasFTApplicationResponse ||
+        !!sfasPTApplicationResponse
+      ) {
+        throw new UnprocessableEntityException(
```

Comment:
This is service layer, hence we must throw error/custom error with the highlighted name so that is it captured and translated appropriately at the controller. 
![image](https://user-images.githubusercontent.com/54600590/161824980-292082c5-33b6-4a6c-9b08-87a7bcc7010e.png)